File: src/brs/web/api/http/handler/SuggestFee.java
package brs.web.api.http.handler;

import brs.feesuggestions.FeeSuggestion;
import brs.feesuggestions.FeeSuggestionCalculator;
import brs.web.api.http.ApiServlet;
import brs.web.api.http.common.LegacyDocTag;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;

import javax.servlet.http.HttpServletRequest;

import static brs.web.api.http.common.ResultFields.*;

public class SuggestFee extends ApiServlet.JsonRequestHandler {

  private final FeeSuggestionCalculator feeSuggestionCalculator;

  public SuggestFee(FeeSuggestionCalculator feeSuggestionCalculator) {
    super(new LegacyDocTag[]{LegacyDocTag.FEES});
    this.feeSuggestionCalculator = feeSuggestionCalculator;
  }

  @Override
  protected
  JsonElement processRequest(HttpServletRequest req) {
    final FeeSuggestion feeSuggestion = feeSuggestionCalculator.giveFeeSuggestion();

    final JsonObject response = new JsonObject();

    response.addProperty(CHEAP_FEE_RESPONSE, feeSuggestion.getCheapFee());
    response.addProperty(STANDARD_FEE_RESPONSE, feeSuggestion.getStandardFee());
    response.addProperty(PRIORITY_FEE_RESPONSE, feeSuggestion.getPriorityFee());

    return response;
  }

}

File: src/main/kotlin/brs/api/http/SuggestFee.kt
package brs.api.http

import brs.api.http.common.ResultFields.CHEAP_FEE_RESPONSE
import brs.api.http.common.ResultFields.PRIORITY_FEE_RESPONSE
import brs.api.http.common.ResultFields.STANDARD_FEE_RESPONSE
import brs.services.FeeSuggestionService
import brs.services.impl.FeeSuggestionServiceImpl
import com.google.gson.JsonElement
import com.google.gson.JsonObject
import javax.servlet.http.HttpServletRequest

internal class SuggestFee(private val feeSuggestionService: FeeSuggestionService) : APIServlet.JsonRequestHandler(arrayOf(APITag.FEES)) {
    override fun processRequest(request: HttpServletRequest): JsonElement {
        val feeSuggestion = feeSuggestionService.giveFeeSuggestion()

        val response = JsonObject()

        response.addProperty(CHEAP_FEE_RESPONSE, feeSuggestion.cheapFee)
        response.addProperty(STANDARD_FEE_RESPONSE, feeSuggestion.standardFee)
        response.addProperty(PRIORITY_FEE_RESPONSE, feeSuggestion.priorityFee)

        return response
    }
}
